addons/mail/static/src/components/activity/activity.js

```diff
@@ -69,7 +70,11 @@ class Activity extends Component {
      * @returns {string}
      */
     get delayLabel() {
-        const today = moment().startOf('day');
+        const userTZ = session.user_context['tz']
+        const todayUTC = moment().utc();
+        // Convert using timezone identifier (e.g., "Europe/Brussels") :
+        const todayUserTZ = moment((new Date(todayUTC.format())).toLocaleString("en-US", {timeZone: userTZ}), "M/D/YYYY, h:m:s A")
+        const today = todayUserTZ.startOf('day');
```

Comment:
I would be more comfortable if you were using something existing here, such as `formatDate` from field utils. If that doesn't do the job, another utils should be made to be (re)used everywhere.
The less logic we have in this component, the better.

Reply:
Unfortunately, `formatDate` is based on locale machine TZ instead of user TZ (from preferences)